server/resourcemanager/operations.go

```diff
@@ -0,0 +1,163 @@
+package resourcemanager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kubeshop/tracetest/server/id"
+	"golang.org/x/exp/slices"
+)
+
+type Operation string
+
+const (
+	OperationList   Operation = "list"
+	OperationCreate Operation = "create"
+	OperationUpdate Operation = "update"
+	OperationGet    Operation = "get"
+	OperationDelete Operation = "delete"
+)
+
+var availableOperations = []Operation{
+	OperationList,
+	OperationCreate,
+	OperationUpdate,
+	OperationGet,
+	OperationDelete,
+}
+
+type SortableHandler interface {
+	SortingFields() []string
+}
+
+type List[T ResourceSpec] interface {
+	SortableHandler
+	List(_ context.Context, take, skip int, query, sortBy, sortDirection string) ([]T, error)
+	Count(_ context.Context, query string) (int, error)
+}
+
+type Create[T ResourceSpec] interface {
+	Create(context.Context, T) (T, error)
+	SetID(T, id.ID) T
+}
+
+type Update[T ResourceSpec] interface {
+	Update(context.Context, T) (T, error)
+}
+
+type Get[T ResourceSpec] interface {
+	Get(context.Context, id.ID) (T, error)
+}
+
+type Delete[T ResourceSpec] interface {
+	Delete(context.Context, id.ID) error
+}
+
+type resourceHandler[T ResourceSpec] struct {
```

Comment:
The idea here was to replace the interface with a struct that can have "optional" functions. This allows for partial resources, like config, to only implement the required methods.

Note that this is an unexported struct. It's mapped when creating a new resourcemanager, so clients don't need to know how this works. Always keep in mind that the idea of this package is to be very flexible, and require as little coupling as possible